include/pm/device.h

```diff
@@ -143,6 +155,42 @@ int pm_device_state_get(const struct device *dev,
 /** Alias for legacy use of device_pm_control_nop */
 #define device_pm_control_nop __DEPRECATED_MACRO NULL
 
+/**
+ * @brief Enable a power management wakeup source
+ *
+ * Enable a wakeup source. This will keep the current device active when the
+ * system is suspended, allowing it to be used to wake up the system.
+ *
+ * @param dev device object to enable.
+ * @param enable @c true to enable or @c false to disable
+ *
+ * @return @c true if the wakeup source was successfully enabled,
+ *         @c false otherwise.
+ */
+bool pm_device_wakeup_enable(struct device *dev, bool enable);
+
+/**
+ * @brief Check if a power management wakeup source is enabled
+ *
+ * Checks if a wake up source is enabled.
+ *
+ * @param dev device object to check.
+ *
+ * @return @c true if the wakeup source is enabled,
+ *         @c false otherwise.
+ */
+bool pm_device_wakeup_is_enabled(const struct device *dev);
+
+/**
+ * @brief Check if a device is wake up capable
+ *
+ * @param dev device object check.
+ *
+ * @return @c true if the device is wake up capable,
+ *         @c false otherwise.
```

Comment:
really ? 
```
faceolin@faceolin-mobl5:~/p/zephyrproject/zephyr$ git grep "@retval" | wc -l
1424
faceolin@faceolin-mobl5:~/p/zephyrproject/zephyr$ git grep "@return " | wc -l
3677
```

Reply:
I mean that `@retval` is used for each return value, eg

```
@retval true x
@retval false y
```

Reply:
gotcha, but @return is more used than @retval. Do we have a guideline or a documentation about which one to use ? Because what you said about @retval being used in most places is not accurate.

Reply:
`@retval` is used when specifying each return code. I can't find any `@return @c` in tree.

Reply:
well, @c can be used anywhere. Most places just use `@return` followed by a value. I'm going to change it since I don't have a hard opinion about it.